src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMapping.java

```diff
@@ -102,8 +107,21 @@ public static LocalDateTime getFirstDateTimeMustBe() {
     public static int getHearingDuration(SscsCaseData caseData) {
         // TODO Adjournments - Check this is the correct logic for Adjournments
         // TODO Future Work - Manual Override
-        // TODO Dependant on SSCS-10116 - Will use Session Category Reference Data
 
+        int duration = getHearingDurationAdjournment(caseData);
+        if (duration > 0) {
+            return duration;
+        }
+
+        duration = getHearingDurationBenefitIssueCodes(caseData);
+        if (duration > 0) {
+            return duration;
+        }
+
+        return DURATION_DEFAULT;
+    }
+
+    public static int getHearingDurationAdjournment(SscsCaseData caseData) {
         if (isNotBlank(caseData.getAdjournCaseNextHearingListingDuration())
                 && caseData.getAdjournCaseNextHearingListingDuration().matches("\\d+")
```

Comment:
Why do we need the regex?

Reply:
to prevent a NumberFormatException by the parseInt, should I remove it and let it fail on that?